vyper/parser/context.py

```diff
@@ -146,8 +146,12 @@ def is_valid_varname(self, name, pos):
 
     # TODO location info for errors
     # Add a new variable
-    def new_variable(self, name, typ, pos=None):
-        if self.is_valid_varname(name, pos):
+    def new_variable(self, name, typ, internal_var=False, pos=None):
+        # mangle internally generated variables so they cannot collide
+        # with user variables.
+        if internal_var:
+            name = '#' + name
```

Comment:
maybe for clarity the name should be mapped to 
```suggestion
            name = '#internal var ' + name
```